compiler/hash-utils/src/graph/mod.rs

```diff
@@ -0,0 +1,67 @@
+//! This module contains graph algorithms and utilities that are used by the
+//! compiler.
+
+pub mod dominators;
+pub mod visit;
+
+use index_vec::Idx;
+
+/// This trait can be used to represent a generic directed
+/// graph. The index of each graph node used a [`index_vec::Idx`].
```

Comment:
Why using indexvec?

Reply:
Since it is most likely that node identifiers will use an `index_vec` to store the nodes.